packages/nextjs/test/performance/client.test.ts

```diff
@@ -5,84 +5,98 @@ import { default as Router } from 'next/router';
 
 import { nextRouterInstrumentation } from '../../src/performance/client';
 
-let readyCalled = false;
+const globalObject = getGlobalObject<
+  Window & {
+    __BUILD_MANIFEST?: {
+      sortedPages?: string[];
+    };
+  }
+>();
+
+const originalBuildManifest = globalObject.__BUILD_MANIFEST;
+const originalBuildManifestRoutes = globalObject.__BUILD_MANIFEST?.sortedPages;
+
 jest.mock('next/router', () => {
-  const router = {};
-  Object.setPrototypeOf(router, { changeState: () => undefined });
+  const eventHandlers: { [eventName: string]: ((...args: any[]) => void)[] } = {};
   return {
     default: {
-      router,
-      route: '/[user]/posts/[id]',
-      readyCallbacks: [],
-      ready(cb: () => void) {
-        readyCalled = true;
-        return cb();
+      events: {
+        on(type: string, handler: (...args: any[]) => void) {
```

Comment:
should we also test if `off` was called on the listener?

Reply:
Added in a340b5c3ce242e28df309bf958f4f49d17597576